src/client.js

```diff
@@ -287,7 +287,16 @@ export function MatrixClient(opts) {
                 _updatePendingEventStatus(room, eventToSend,
                                           EventStatus.SENDING);
             }
-            return _sendEventHttpRequest(self, eventToSend);
+            let promise = _sendEventHttpRequest(self, eventToSend);
```

Comment:
nit: you could make this callback async, then you can just await here and don't need the promise reassignment below.